### DISCUSSION TITLE
Rebuild quando aparece o teclado

### DISCUSSION THREAD

**Discussion by bmcandido** [Flutterando/forum#181] [Perguntas e Respostas]
Estou com um problema no meu app em relação ao teclado, toda vez que ele aparece na tela aconteceu um rebuild, em qualquer lugar do meu aplicativo que eu o chame acontece o problema, como resolver?

**Comment by maurojohann** [comment]
Quando você toca no TextField, o teclado aparece. E quando o teclado aparece, o tamanho da tela muda. Isso faz com que o flutter entenda que é preciso chamar o metodo build(), é o Flutter quem descide se deve ou nao reconstruir a tela. Se voce tem uma logica que roda sempre que o metodo build é reconstruido, tente mover isso para o initState().
